src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java

```diff
@@ -61,9 +61,11 @@
 
 public abstract class EntityMock extends Entity.Spigot implements Entity, MessageTarget
 {
+	private static int idIndex = 0;
```

Comment:
This should be atomic to prevent any weird concurrency issues, which are unlikely to happen, but possible. `ENTITY_COUNTER` is also a clearer name than `idIndex`.
```suggestion
{

	private static final AtomicInteger ENTITY_COUNTER = new AtomicInteger();